lib/kernel/src/os.erl

```diff
@@ -313,7 +313,15 @@ mk_cmd(_,Cmd) ->
     %% We use an absolute path here because we do not want the path to be
     %% searched in case a stale NFS handle is somewhere in the path before
     %% the sh command.
-    {"/bin/sh -s unix:cmd", [out],
+    Command = case {filelib:is_regular("/bin/sh"),
```

Comment:
`mk_cmd` can be called in a loop, so I think it makes sense to optimize this code a bit. If this is only for Android, then I would prefer if we could have a branch similar to windows for Android. That is:

```erlang
mk_cmd({win32,Wtype},Cmd) ->
  ...
mk_cmd({unix,android},Cmd) ->
  mk_unix_cmd("/system/bin/sh");
mk_cmd(_,Cmd) ->
  mk_unix_cmd("/bin/sh").
```

If it is not only for Android, then break it up into two case statements, so that only one file access has to be done in the normal case. That is:

```erlang
case filelib:is_regular("/bin/sh") of
  true -> "/bin/sh";
  false -> case filelib:is_regular("/system/bin/sh") of
             true -> "/system/bin/sh";
             false -> "/bin/sh";
           end.
```

Reply:
> `mk_cmd` can be called in a loop, so I think it makes sense to optimize this code a bit.

Sure, let me give it a try.

> If this is only for Android, then I would prefer if we could have a branch similar to Windows for Android. That is:
> 
> ```erlang
> mk_cmd({unix,android},Cmd) ->
>   mk_unix_cmd("/system/bin/sh");
> ```

That's exactly what I had in mind at first but `os:type()` returns `{unix,linux}` on Android. In fact, this is expected as the documentation states it gets the Osname part from `uname -s`, which indeed returns `Linux` on Android:
```
$ uname -s
Linux
```

> so that only one file access has to be done in the normal case. That is:
> 
> ```erlang
> case filelib:is_regular("/bin/sh") of
>   true -> "/bin/sh";
>   false -> case filelib:is_regular("/system/bin/sh") of
>              true -> "/system/bin/sh";
>              false -> "/bin/sh";
>            end.
> ```

I've re-implemented it this way to have only one file access as suggested. Let me know if you see any other improvements to take into account.